### PR TITLE
[SPARK-28538][UI] Document SQL page

### PR DESCRIPTION
**Pull request by zhengruifeng**
## What changes were proposed in this pull request?
1, add basic doc for each page;
2, doc SQL page with an exmple;

## How was this patch tested?
locally built

图片

图片

### CODE REVIEW

**Comment by zhengruifeng** [comment]
ping @gatorsmile

File: docs/monitoring.md [hunk 1]
```diff
@@ -40,6 +40,100 @@ To view the web UI after the fact, set `spark.eventLog.enabled` to true before s
...
+
+We can see that detailed information of each stage. The first block 'WholeStageCodegen'  
+compile multiple operator ('LocalTableScan' and 'HashAggregate') together into a single Java
+function to improve performance, and metrics like number of rows and spill size are listed in
+the block. The second block 'Exchange' shows the metrics on the shuffle exchange, including
+number of written shuffle records, total data size, etc.
+
+
+<p style="text-align: center;">
+  <img src="img/webui-sql-plan.png"
```
Comment (dongjoon-hyun) [inline docs/monitoring.md]: Hi, @srowen .
Do we need to ask @zhengruifeng the source for this `png`?

File: docs/monitoring.md [hunk 2]
```diff
@@ -40,6 +40,100 @@ To view the web UI after the fact, set `spark.eventLog.enabled` to true before s
 application. This configures Spark to log Spark events that encode the information displayed
 in the UI to persisted storage.
 
+## Web UI Tabs
+The web UI provides an overview of the Spark cluster and is composed of following tabs:
+
+### Jobs Tab
+The Jobs tab displays a summary page of all jobs in the Spark application and a detailed page
```
Comment (gatorsmile) [inline docs/monitoring.md]: Instead doing everything in a huge "Monitoring and Instrumentation", could you create a separate page for each one and add the links to these contents? In the future, we can adding more descriptions in these pages.

File: docs/monitoring.md (hunk 1 above)
Comment (zhengruifeng) [reply docs/monitoring.md]: HI, @dongjoon-hyun  I local run the example and make screenshots to png files.

**Comment by zhengruifeng** [comment]
图片

图片

图片
图片
图片
图片

**Comment by zhengruifeng** [comment]
@gatorsmile  I make `Web UI` a seperate page.
I also attempt to add a link in `More`, after `Montoring`. However, it did not work. Do you know how to add it?

File: docs/web-ui.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+page, you see the detailed page for that job. The detailed page further shows the event timeline,
+DAG visualization, and all stages of the job.
+
+## Stages Tab
+The Stages tab displays a summary page that shows the current state of all stages of all jobs in
+the Spark application, and, when you click on a stage, a detailed page for that stage. The details
+page shows the event timeline, DAG visualization, and all tasks for the stage.
+
+## Storage Tab
+The Storage tab displays the persisted RDDs, if any, in the application. The summary page shows
```
Comment (srowen) [inline docs/web-ui.md]: We might say "RDDs and DataFrames" although the page refers to RDDs I believe

File: docs/web-ui.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+**Table of Contents**
+
+* This will become a table of contents (this text will be scraped).
+{:toc}
+
+## Jobs Tab
+The Jobs tab displays a summary page of all jobs in the Spark application and a detailed page
+for each job. The summary page shows high-level information, such as the status, duration, and
+progress of all jobs and the overall event timeline. When you click on a job on the summary
+page, you see the detailed page for that job. The detailed page further shows the event timeline,
```
Comment (srowen) [inline docs/web-ui.md]: should detailed be details?

**Review by srowen — COMMENTED** [review]
Looks useful as a basic explanation of the UI tabs

File: docs/monitoring.md [hunk 5]
```diff
@@ -23,7 +23,7 @@ There are several ways to monitor Spark applications: web UIs, metrics, and exte
 
 # Web Interfaces
 
-Every SparkContext launches a web UI, by default on port 4040, that
+Every SparkContext launches a [Web UI](web-ui.html), by default on port 4040, that
```
Comment (zhengruifeng) [inline docs/monitoring.md]: I want to add a `Web UI` link behind `Monitoring`, however it never work. 

图片

Dou you know how to add it? @srowen @gatorsmile

**Comment by zhengruifeng** [comment]
retest this please

File: docs/monitoring.md (hunk 5 above)
Comment (srowen) [reply docs/monitoring.md]: I think it's in `_layouts/global.html`, but I am not sure this needs a top level menu item.

Comment (zhengruifeng) [reply docs/monitoring.md]: @gatorsmile What's your thought? Do we need a top level menu item?

Comment (gatorsmile) [reply docs/monitoring.md]: Either is fine to me. I majorly care the contents after we have the basic UI descriptions. :-)

File: docs/web-ui.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+<p style="text-align: center;">
+  <img src="img/webui-sql-plan.png"
+       title="logical plans and the physical plan"
+       alt="logical plans and the physical plan"
+       width="80%" />
+  <!-- Images are downsized intentionally to improve quality on retina displays -->
+</p>
+Clicking the 'Details' link on the bottom displays the logical plans and the physical plan, which
+illustrate how Spark parse, analyze, optimize and perform the query.
```
Comment (kiszk) [inline docs/web-ui.md]: `how Spark parse, analyze, optimize and perform the query.` -> `how Spark parses, analyzes, optimizes and performs the query.`

**Comment by srowen** [comment]
@zhengruifeng just checking if you want to make more changes or it's ready, just because there is ` [WIP]` in the title.

**Comment by zhengruifeng** [comment]
@srowen It should be ready. Thanks for reviewing!

**Review by srowen — APPROVED** [review]

**Pull request closed without merging**